src/rgw/rgw_arrow.cc

```diff
@@ -0,0 +1,97 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+
+#include <iostream>
+#include <mutex>
+#include <map>
+
+#include "arrow/type.h"
+#include "arrow/flight/server.h"
+
+#include "common/dout.h"
+
+#include "rgw_arrow.h"
+
+
+#define dout_subsys ceph_subsys_rgw
```

Comment:
should we consider a new component here?

Reply:
I've made it its own component. Thanks!